src/zope/pluggableauth/plugins/idpicker.py

```diff
@@ -53,11 +59,11 @@ class IdPicker(NameChooser):
     """
     def chooseName(self, name, object):
         i = 0
-        name = unicode(name)
+        name = text_type(name)
         orig = name
         while (not name) or (name in self.context):
             i += 1
-            name = orig+str(i)
+            name = orig + str(i)
```

Comment:
"where `text_type` is not `str`" == Python 2, yes?
```
$ python2
>>> u'\N{SNOWMAN}-' + str(1)
u'\u2603-1'
```

str(number) gives pure-ASCII strings which can be automatically promoted to unicode on Python 2.